### DISCUSSION TITLE
Right way to warm/hot start CLP

### DISCUSSION THREAD

**Discussion by fontanf** [coin-or/Clp#205] [Q&A] [answered]
Hi,

I have a first mode which I solve with CLP.
I would like to solve variants of the problem for which only the bounds change. These new problems are in different `Clp_Simplex` structures which have the same size and the same variable ordering as the initial `Clp_Simplex` structure. What is the right way to extract the warm/hot start information from the initial `Clp_Simplex` and pass it to the new `Clp_Simplex` structures?
I see a `CoinWarmStartBasis * 	getBasis () const` method in the documentation https://www.coin-or.org/Doxygen/Clp/classClpSimplex.html but no `setBasis` method.

I would also like to solve variants with different objectives or additional constraints. Is there something useful that can be done to warm/hot start Clp in this case?

**Comment by jjhforrest** [comment]
ClpSimplex.?pp - writeBasis and readBasis.

**Reply by fontanf** [reply]
These methods will cause disk accesses. I would have expected some C/C++ structures to store rather than files to write. Is the best solution to copy the `ClpSimplex` object with the copy constructor and then to modify it? In this case, I can't find the copy constructor in the C interface https://www.coin-or.org/Doxygen/Clp/Clp__C__Interface_8h.html

**Comment by jjhforrest** [comment]
With C++

If just bounds changing (any way - no change in number rows/columns) -

ClpSimplex alreadySolved and ClpSimplex modified

modified.moveInfo(&alreadySolved); // If just bounds then could be 
,true) just to do basis status

Will work with different objectives.

Not experienced using C interface.

**Comment by jjhforrest** [comment, marked as answer]
Looking at Clp_C_Interface.h

Clp_Simplex * alreadySolved and Clp_Simplex *modified

Clp_copyinStatus(modified, Clp_statusArray(alreadySolved));

**Reply by fontanf** [reply]
Thanks, it seems to be what I was looking for. With the status array, I don't need to have both ClpSimplex in memory at the same time
